src/notes/notes.js

```diff
@@ -0,0 +1,352 @@
+import { Record } from "../backend-storage/record-class.js";
+import RecordsStorage from "../backend-storage/records-api.js";
+
+// id of the most recently accessed (clicked, edited, etc.) note element
+let CURRENT_NOTE_ID = null;
+
+/*
+Loads note records from storage and creates note-elements to display them in notes.html
+Parameters: None
+Returns: None
+*/
+function loadAllNotesFromStorage() {
+    // get container where notes are displayed
+    const notesDisplay = document.querySelector(".notes-display");
+    // retrieve notes from local storage
+    const notesList = RecordsStorage.getAllRecords("note");
+    // loop through retrieved notes, creating a note element for each one
+    for (const noteRecord of notesList) {
+        // create and initialize new note element
+        let noteElem = document.createElement("note-element");
+        noteElem.id = noteRecord.id;
+        noteElem.title = noteRecord.title;
+        noteElem.date = noteRecord.created;
+        noteElem.content = noteRecord.field1;
+        // add note element to html page
+        notesDisplay.prepend(noteElem);
+        // add listeners to note element that check if it is clicked to update it
+        _addListeners(noteElem);
+    }
+}
+
+// submit to local storage
+function submitToStorage() {
+    // the title of the note
+    const noteTitle = document.getElementById("note-editor-title");
+    const noteTitleVal = noteTitle.value;
+    // the textbox to enter notes in
+    const noteContent = document.getElementById("note-editor-content");
+    const noteContentVal = noteContent.value;
+    // create a new record to store the note
+    const noteRecord = new Record("note", {
+        field1: noteContentVal,
+        title: noteTitleVal,
+        id: CURRENT_NOTE_ID, // will be null if creating one, or a value if updating
+    });
+    // in local storage, update note record if it exists, otherwise create a new one
+    if (CURRENT_NOTE_ID) {
+        RecordsStorage.updateRecord(noteRecord);
+    } else {
+        RecordsStorage.createRecord(noteRecord);
+    }
+}
+
+/*
+Given a note id, delete the note from local storage and remove it from the html page
+Parameters:
+    - noteId: string or int of a note's id
+Returns: None
+*/
+function deleteFromStorage(noteId) {
+    // Comes in as string, so we convert to a Number
+    RecordsStorage.deleteRecord(Number(noteId));
```

Comment:
ParseInt() vs Number()? I don't think there's a more correct decision, so whichever is fine, as long as we are consistent